### PR TITLE
Resolvingissues

### PR DESCRIPTION
**Pull request by KrzysztofDzioba**

### CODE REVIEW

File: server/src/main/java/com/battleships/PlayerCommand.java [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+import java.util.Optional;
+
 class PlayerCommand {
     private final Command type;
-    private final String value;
+    private Optional<String> valueOptional;
 
     PlayerCommand(String playerInput) {
         // TODO validate it !
         String[] inputCommand = playerInput.split(CommunicatingProtocol.getSeparator());
```
Comment (JaroslawZaczyk) [inline server/src/main/java/com/battleships/PlayerCommand.java:11]: dot train. Extract separator before using it as split method parameter.

File: client/src/main/java/com/battleships/start_window/connection/ServerIO.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    ServerIO(PrintWriter socketWriter, Scanner socketScanner) {
+        this.socketWriter = socketWriter;
+        this.socketScanner = socketScanner;
+    }
+
+
+    Optional<String> getMessageOptional() {
+        Optional <String> messageOptional = Optional.empty();
+        if (socketScanner.hasNextLine()) {
+            messageOptional = Optional.of(socketScanner.nextLine());
```
Comment (JaroslawZaczyk) [inline client/src/main/java/com/battleships/start_window/connection/ServerIO.java:20]: You should extract result of nextLine method before using it. For a clean code and the other great stuff.

File: client/src/main/java/com/battleships/start_window/connection/Connection.java [hunk 3]
```diff
@@ -115,4 +113,43 @@ public void sendToServer(Command command, String commandValue) {
...
+            logger.error(LogMessages.CANNOT_OBTAIN_SOCKET_OUTPUSTREAM);
+        }
+        return outputStreamOptional;
+    }
+
+    private Optional<InputStream> getInputStreamOptional() {
+        Optional <InputStream> inputStreamOptional = Optional.empty();
+        try {
+            if (socket.isPresent()) {
+                inputStreamOptional = Optional.of(socket.get().getInputStream());
```
Comment (JaroslawZaczyk) [inline client/src/main/java/com/battleships/start_window/connection/Connection.java:148]: the same with train

**Review by JaroslawZaczyk — APPROVED** [review]
Looks good.

File: client/src/main/java/com/battleships/start_window/connection/Connection.java [hunk 4]
```diff
@@ -7,18 +7,21 @@
...
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
 import java.io.PrintWriter;
+import java.net.InetSocketAddress;
 import java.net.Socket;
 import java.util.Optional;
 import java.util.Scanner;
 
 public enum Connection {
```
Comment (totine) [inline client/src/main/java/com/battleships/start_window/connection/Connection.java:18]: Maybe change order of methods - connect first, disconnect later

**Review by totine — APPROVED** [review]

File: server/src/main/java/com/battleships/PlayerCommand.java (hunk 1 above)
Comment (dmuca) [reply server/src/main/java/com/battleships/PlayerCommand.java:11]: extracted

File: client/src/main/java/com/battleships/start_window/connection/Connection.java (hunk 3 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/connection/Connection.java:148]: Cleaned

File: client/src/main/java/com/battleships/start_window/connection/Connection.java (hunk 4 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/connection/Connection.java:18]: Changed : )

File: client/src/main/java/com/battleships/start_window/connection/ServerIO.java (hunk 2 above)
Comment (dmuca) [reply client/src/main/java/com/battleships/start_window/connection/ServerIO.java:20]: Extracted, we want clean code and the other great stuff,

**Pull request merged**